common/beerocks/bwl/dummy/base_wlan_hal_dummy.cpp

```diff
@@ -280,6 +280,24 @@ bool base_wlan_hal_dummy::set(const std::string &param, const std::string &value
 
 bool base_wlan_hal_dummy::ping() { return true; }
 
+bool base_wlan_hal_dummy::write_status_file(const std::string &filename, const std::string &value)
+{
+    auto full_path{get_status_dir() + "/" + filename};
+    auto full_path_tmp{full_path + ".tmp"};
+    std::ofstream statusfile(full_path_tmp);
+    statusfile << value;
+    if (!statusfile) {
+        LOG(ERROR) << "Failed writing to " << full_path;
+        return false;
+    }
+    statusfile.close();
+    if (rename(full_path_tmp.c_str(), full_path.c_str()) < 0) {
```

Comment:
On failure, a nonzero value is returned.
```suggestion
    if (rename(full_path_tmp.c_str(), full_path.c_str()) != 0) {
```

Reply:
`man 2 rename`

```
RETURN VALUE
       On success, zero is returned.  On error, -1 is returned, and errno is set appropriately.
```